### ISSUE TITLE
First station name not visible while entering categories

### ISSUE THREAD

**Issue by TheBossME** [coffeegreg/YTuner#7]
No name visible when entering submenus 1st station (name not there, only blank) but playable. This appear only on a small amount of countries like Finland, Bosnia, Italia, Poland, Slovakia.... etc.

British Indian Ocean Territory access do freeze my Yamaha DSP-Z7.

Improvement: sorting name, ascending, descending 

![image](https://github.com/coffeegreg/YTuner/assets/72735184/4664a82e-1042-47de-a184-f403560dd858)

**Comment by coffeegreg** [comment]
This issue should be releated to radio-browser.info project. :wink: But seriously ... Our AVRs do not correctly support all characters, especially those specific to a given country or other called as escaping special characters, and unfortunately we will not change this. For example, my Yamaha AVRs do not handle all Polish characters correctly and display "_" character instead. The same thing happens with other languages. First of all, we have no influence on how the station will be saved in the radio-browser.info database and we can not modify AVR firmware to support additional characters too. :thinking:

Let's take a look at your case with Bosnia and Herzegovina list stations.
Radio-browser.info API returns something like this :
```
...
serveruuid	null
name	"\tRadio Tešanj 92.2MHz"
url	"http://radio.daj.ba:8082/stream"
...
```
as a part of JSON structure of first station.
As we can see there is some unexpected charakters like `\t` in name field which are interpreted as `TAB` character. Unfortunately, Yamaha AVRs do not like `TAB` character and display empty name in this place.
I suspect a similar problem will occur when appear : `\n`,`\b`,`\f`,`\r`,`\\`. Nevertheless, I'll try to fix it.

Second case.
I didn't find `British Indian Ocean Terorrity` :wink:  in my AVR country list menu, but I've found `British Indian Ocean Territory` and there are some "Tick Tock Radio" stations. You have made a typo or someone else who enter this "Terorrity" :wink:  value to radio-browser.info country database made it.
```
Improvement: sorting name, ascending, descending
```
This will be easy task but will again expand the number of options in the `ytuner.ini` file.

Thanks for finding more things to improve. :+1:

**Comment by coffeegreg** [comment]
Hi Beatrice,

1. Don't worry, I remember about the filtering functionality. Not only by `codec` but also by `country` and `language`. I don't want to take a look to ycast advanced now. As I mentioned before I'll try to come up with something myself. Looking at other ideas kills my creativity. :wink: I also remember that ycast always had one configuration for all AVR devices connected to it. YTuner "inherited" this "defective" concept when I was reviewing the ycast code and thinking about coding YTuner. The effect is that the content of the `stations.ini/yaml` file is the same for all AVR devices that connect to it. If someone has only one AVR, there is no problem, but for me it is. Therefore, the `Bookmark` functionality in YTuner, which is not available in ycast, was created on the basis of resource sharing (one `bookmark.xml`) or its personalization per AVR device (each AVR will own its own bookmark file).
Now I plan to change the functionality related to "My Stations" so that the concept is analogous to `Bookmark`
Finally, a similar idea should be implemented with filtering. My RX-V671 doesn't support AAC, but the RX-V673 does.
Give me some time because I've had some other activities and work lately and I need to deal with them. 

2. Yes. Of course, there is such functionality of the `radio-browser.info` API. Moreover, there is a search function implemented in YTuner, but I have never tested it because my AVR does not support it and I am afraid your Yamaha DSP-Z7 does not support it either. :worried:

Regards
Greg

**Comment by TheBossME** [comment]
I do understand. I was also a creative person in the past life. 
the filter yaml file would be an example only. Programming that feature is an other thing. 

Forgave me when I do pushing you. That's not what I liked to do.

Do other things first. The music hobby must stand behind all other things.

Everytime nice communication with you. Love that 😘

Bea

**Comment by coffeegreg** [comment]
Now, I can take a look fo ycast advanced... :eyeglasses: 
Well, I think YTuner filtering is almost the same or even more advanced (e.g. URL filtering, filtering with wildcards) :thinking: 
I hope you will be pleased with the new capabilities of YTuner. :+1: 
Greg

**Comment by coffeegreg** [comment]
Bug releated to _"First station name not visible while entering categories"_ has been fixed with version 1.1.0.
Filtering and sorting features added (due this (https://github.com/coffeegreg/YTuner/issues/11) issue solution need another approach to solve the problem with selected AVRs).
